cluster/log-dump/log-dump.sh

```diff
@@ -276,24 +348,31 @@ function dump_nodes() {
     node_logfiles_all="${node_logfiles_all} ${hollow_node_logfiles}"
   fi
 
-  nodes_selected_for_logs=()
+  linux_nodes_selected_for_logs=()
   if [[ -n "${LOGDUMP_ONLY_N_RANDOM_NODES:-}" ]]; then
     # We randomly choose 'LOGDUMP_ONLY_N_RANDOM_NODES' many nodes for fetching logs.
     for index in `shuf -i 0-$(( ${#node_names[*]} - 1 )) -n ${LOGDUMP_ONLY_N_RANDOM_NODES}`
     do
-      nodes_selected_for_logs+=("${node_names[$index]}")
+      linux_nodes_selected_for_logs+=("${node_names[$index]}")
     done
   else
-    nodes_selected_for_logs=( "${node_names[@]}" )
+    linux_nodes_selected_for_logs=( "${node_names[@]}" )
   fi
+  all_selected_nodes=( "${linux_nodes_selected_for_logs[@]}" )
+  all_selected_nodes+=( "${windows_node_names[@]}" )
 
   proc=${max_dump_processes}
-  for node_name in "${nodes_selected_for_logs[@]}"; do
+  for i in "${!all_selected_nodes[@]}"; do
```

Comment:
This was a new one to me, expands to list of array indices.  Took me a while to grok this was how you're gating "should I treat this as a linux node or a windows node".

Reply:
Yeah, I also learned this syntax for the first time.

The tricky part here was trying to preserve / take advantage of the max_dump_processes code below. This necessitated using just a single loop, so appending the Windows nodes after the Linux nodes in this array and using the index to tell the difference was the cleanest approach I could come up with.

Alternatively we could iterate over the Linux and Windows node arrays separately, but I didn't see an obvious way to update the max_dump_processes part so that all the log saving would happen in parallel. I can look into this further or I'm open to suggestions if desired.